### ISSUE TITLE
VF2pp tests are not passing during the regularly scheduled randomly ordered tests 

### ISSUE THREAD

**Issue by dschult** [networkx/networkx#5992]
I think we decided that this is due to some tests changing the graph stored in a test class attribute.
@kpetridis24 has some ideas for how that would get solved.
It'd be good to get this merged soon to avoid these failures, so we should put it in a separate PR.

**Comment by kpetridis24** [comment]
@dschult oops, i forgot about that while i was fixing everything else. I will have it fixed during the next days and i will tag you when i push the changes!

**Comment by rossbar** [comment]
Thanks @kpetridis24 - a couple tips for this:
 1. Go ahead and submit these fixups as their own PR against main instead of adding them to an already-open PR. We want to get these fixes in as soon as possible, so having them be self-contained will help!
 2. There is no CI job that runs the tests in random order *on PR/pushes* - only a scheduled job - which means that the changes won't be evaluated properly in CI. To verify the fix works, you can `pip install pytest-randomly` locally and run the test suite locally a few times.

**Comment by kpetridis24** [comment]
@dschult @rossbar I accidentally didn't include the changes in a new PR... The issue was that the graphs were created as a class member variable and were accessed/modified by the unit tests
```python
class Test:
  G1 = Graph
  G2 = Graph
  
  def test1():
    modify G1, G2

  def test2():
    modify G1, G2
```

The ONLY change that i made was

```python
class Test:
  def test1():
    G1 = Graph
    G2 = Graph
    modify G1, G2

  def test2():
    G1 = Graph
    G2 = Graph
    modify G1, G2
```
So now no test depends on the one running prior to it.

**Comment by kpetridis24** [comment]
There are 2 doc-related failures that i cannot understand.

**Comment by rossbar** [comment]
I believe this was also fixed by #5972
